nectar/src/config/settings.rs

```diff
@@ -20,19 +20,53 @@ pub struct Settings {
 
 #[derive(Clone, Debug, PartialEq)]
 pub struct Bitcoin {
-    pub network: bitcoin::Network,
+    pub network: ledger::Bitcoin,
     pub bitcoind: Bitcoind,
 }
 
-impl Default for Bitcoin {
-    fn default() -> Self {
+impl Bitcoin {
+    pub fn new(network: ledger::Bitcoin) -> Self {
         Self {
-            network: bitcoin::Network::Regtest,
-            bitcoind: Bitcoind {
-                node_url: Url::parse("http://localhost:18443")
-                    .expect("static string to be a valid url"),
-            },
+            network,
+            bitcoind: Bitcoind::new(network),
+        }
+    }
+
+    fn from_file(bitcoin: file::Bitcoin, comit_network: Option<comit::Network>) -> Result<Self> {
+        if let Some(comit_network) = comit_network {
+            let inferred = ledger::Bitcoin::from(comit_network);
+            if inferred != bitcoin.network {
+                anyhow::bail!(
+                    "inferred Bitcoin network {} from CLI argument {} but config file says {}",
```

Comment:
For the moment, I am just mirroring the functionality of cnd here. The reason I am a bit hesitant to change this to "expected" is because I am not quite clear, what the authoritative source is hence I am just erroring because it is not consistent.

If we agree on what the authoritative source is (config file or CLI argument), then we can remove this check and just always roll with that.

Reply:
Also, we have to take into account that some of these values may come from defaults. This check ONLY fires if the user specifies a COMIT network on the commandline AND specified something the config file.

If any of those are missing, we take what the user specified. If they want to connect to mix networks, they will have to configure that explicitly in the config file. That should work I believe.